### PR TITLE
Swap out sift3 for swift4 common

### LINKED ISSUE #113: Sidrite has a newer version of Sift string distance algorithm
http://siderite.blogspot.com/2014/11/super-fast-and-accurate-string-distance.html

### PR DESCRIPTION
**Pull request by alberthu16**
Fix #113

From [Siderite's blog](https://siderite.blogspot.com/2014/11/super-fast-and-accurate-string-distance.html): 

> After implementing Sift4, I can now tell you that the simple version is slightly better than Sift3 for small maxOffset values like 5, but it gets better as the value increases. The common version is a bit more complex, but the error decreases with 33% and maintains a low error for large maxOffset values.

I pretty much just copied over the source. I took out the `maxDistance` for simplicity, and set a default `maxOffset` of 5 for consistency.

### CODE REVIEW

**Comment by alberthu16** [comment]
@derrickko any thoughts on this?

**Comment by derrickko** [comment]
@alberthu16 looks awesome. Did you update the tests to make sure they pass?

**Comment by alberthu16** [comment]
Thanks @derrickko! Actually, I didn't have to update the tests to make sure they pass. None of them broke, so I'm assuming the functionality is consistent. Should I add some more domain checks? There are no tests for the actual edit distance algorithm currently.

**Comment by alberthu16** [comment]
@derrickko thoughts on adding tests or leaving as is?

File: src/mailcheck.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ var Mailcheck = {
     opts.domains = opts.domains || Mailcheck.defaultDomains;
     opts.secondLevelDomains = opts.secondLevelDomains || Mailcheck.defaultSecondLevelDomains;
     opts.topLevelDomains = opts.topLevelDomains || Mailcheck.defaultTopLevelDomains;
-    opts.distanceFunction = opts.distanceFunction || Mailcheck.sift3Distance;
+    opts.distanceFunction = opts.distanceFunction || Mailcheck.sift4Distance;
```
Comment (alberthu16) [inline src/mailcheck.js]: this is horrible naming!!!!!

**Pull request merged**